### PR TITLE
move mapToStruct to marshal package

### PR DESCRIPTION
**Pull request by ElecTwix**
added interface case to mapToStruct
added unit tests

### CODE REVIEW

File: pkg/marshal/marshal_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+func TestUnMarshalMapToStruct(t *testing.T) {
+	testkey := "testKey"
+	testDataMap := make(map[string]interface{})
+
+	// nil data
+	testDataMap[testkey] = nil
+
+	testObj := new(testMarshalStruct)
+	err := marshal.UnMarshalMapToStruct(testDataMap, testObj)
+	if err == nil {
```
Comment (phughk) [inline pkg/marshal/marshal_test.go:25]: testify require please 😄

File: pkg/marshal/marshal_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+	testDataMap["testInt"] = testObj.TestInt
+	testDataMap["testBool"] = testObj.TestBool
+	testDataMap["testString"] = testObj.TestString
+	testDataMap["testAny"] = testObj.TestAny
+
+	delete(testDataMap, testkey)
+
+	err = marshal.UnMarshalMapToStruct(testDataMap, testObj)
+	if err != nil {
```
Comment (phughk) [inline pkg/marshal/marshal_test.go:44]: Also testify 🙏

**Review by phughk — COMMENTED** [review]
Amazing thank you! Minor comment but otherwise can land. Adding hacktoberfest-accepted tag

**Comment by phughk** [comment]
Hey, I can see the temptation to change from nightly to stable @ElecTwix . There is a real possibility it could have solved the build. But I would quite like the build to be tied to nightly. That way we are certain there is compatibility with latest changes coming from the db.
Could you please revert the docker change?

**Comment by ElecTwix** [comment]
I will test out with my own repo I don't wanna create buch of commit in official repo.

**Comment by naisofly** [comment]
@ElecTwix Sorry I missed this!
Thank you for contributing to SurrealDB during Hacktoberfest 2023 🎉
Kindly email [community@surrealdb.com](mailto:community@surrealdb.com) to coordinate the delivery of your SurrealDB swag! 😁

**Comment by ElecTwix** [comment]
> @ElecTwix Sorry I missed this! Thank you for contributing to SurrealDB during Hacktoberfest 2023 🎉 Kindly email [community@surrealdb.com](mailto:community@surrealdb.com) to coordinate the delivery of your SurrealDB swag! 😁

Hi @naisofly,

I emailed community@surrealdb.com on Fri, Dec 1 to coordinate the delivery, but I'm happy to resend it if it didn't make it through. Please let me know the next steps I should take to claim my prize.

Thank you again for your time and I look forward to hearing from you soon!

**Pull request closed without merging**